src/uct/tcp/tcp_ep.c

```diff
@@ -615,17 +619,22 @@ static inline unsigned uct_tcp_ep_recv(uct_tcp_ep_t *ep, size_t recv_length)
 static unsigned uct_tcp_ep_progress_data_tx(uct_tcp_ep_t *ep)
 {
     unsigned count = 0;
+    ssize_t ret;
     size_t sent_length;
 
     ucs_trace_func("ep=%p", ep);
 
     if (uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
-        if (!(ep->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_ZCOPY_TX))) {
-            count += uct_tcp_ep_send(ep, &sent_length);
-        } else {
-            count += uct_tcp_ep_sendv(ep, &sent_length);
+        ret = (!(ep->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_ZCOPY_TX)) ?
+               uct_tcp_ep_send(ep, &sent_length) :
+               uct_tcp_ep_sendv(ep, &sent_length));
+        if (ucs_unlikely(ret < 0)) {
+            //uct_tcp_ep_handle_disconnected(ep, &ep->tx);
```

Comment:
leftover or useful comment? pls comment in c-style and add `TODO: ` if it's something useful

Reply:
leftover - fixed